### PR TITLE
Verify block sig

### PR DESCRIPTION
**Pull request by shayzluf**
[Part of] #5116

---

# Description
In order to keep slasher detect slashing and not store validator keys we need to validate blocks in beacon chain before adding them to block stream.

### CODE REVIEW

File: beacon-chain/db/kv/blocks.go [hunk 1]
```diff
@@ -354,6 +354,20 @@ func (k *Store) GenesisBlock(ctx context.Context) (*ethpb.SignedBeaconBlock, err
 	return block, err
 }
 
+// GenesisBlockRoot retrieves the genesis block root from db.
+func (k *Store) GenesisBlockRoot(ctx context.Context) ([32]byte, error) {
```
Comment (nisdas) [inline beacon-chain/db/kv/blocks.go]: we dont need this please revert this change.

File: beacon-chain/rpc/beacon/blocks.go [hunk 2]
```diff
@@ -191,6 +193,28 @@ func (bs *Server) StreamBlocks(_ *ptypes.Empty, stream ethpb.BeaconChain_StreamB
...
+					return err
+				}
+				proposer, err := beaconState.ValidatorAtIndex(data.SignedBlock.Block.ProposerIndex)
+				if err != nil {
+					return err
+				}
+
+				// Verify proposer signature before sending to stream.
+				currentEpoch := helpers.SlotToEpoch(beaconState.Slot())
+				domain, err := helpers.Domain(beaconState.Fork(), currentEpoch, params.BeaconConfig().DomainBeaconProposer, genesisRoot[:])
```
Comment (nisdas) [inline beacon-chain/rpc/beacon/blocks.go]: the spec says genesis validator root, not genesis block root. You can get it from the state

Comment (shayzluf) [reply beacon-chain/rpc/beacon/blocks.go]: thanks, will do

File: beacon-chain/db/kv/blocks.go (hunk 1 above)
Comment (shayzluf) [reply beacon-chain/db/kv/blocks.go]: reverted. thanks

File: beacon-chain/rpc/beacon/blocks.go (hunk 2 above)
Comment (shayzluf) [reply beacon-chain/rpc/beacon/blocks.go]: changed
